nestkernel/connector_base.h

```diff
@@ -467,10 +467,7 @@ class Connector : public ConnectorBase
   }
 
   size_t
-  find_first_target( const size_t tid,
-    const size_t start_lcid,
-    const size_t target_node_id,
-    bool use_compressed_spikes = false ) const override
+  find_first_target( const size_t tid, const size_t start_lcid, const size_t target_node_id ) const override
   {
     size_t lcid = start_lcid;
```

Comment:
This function should only be used when `use_compressed_spikes` is enabled.
```suggestion
  find_first_target( const size_t tid, const size_t start_lcid, const size_t target_node_id ) const override
  {
    assert(use_compressed_spikes())
    size_t lcid = start_lcid;
```

Reply:
Good point! Please ignore the first version of this comment—clearly we need the accessor method here. I will add spaces to make our format checker happy.

Reply:
Sorry, you need to access it through `kernel().connection_manager`, I guess.

Reply:
And because we are in a header, we then risk circular-inclusion hell. Since the function is only called from a single place and that place is assert-protected, this is not a major risk. Once #3544 is merged to avoid the kernel-include problems we have, we can revisit this particular point (I marked it as TODO in the code). And #3544 needs to wait for PyNEST-NG, so it could be a few months.